jiraServerPlugin/src/main/java/com/opsgenie/plugin/service/OpsgeniePluginSettingsManagerImpl.java

```diff
@@ -50,43 +53,54 @@ public void setServerId(String serverId) {
-        //plugin settings may not be sync with the user existence. If user deleted then the customer need to reinstall the plugin in order to add new user :((
-        if (applicationUser == null && StringUtils.isNotBlank(apiKey) && StringUtils.isNotBlank(baseUrl)) {
-            UserDetails userDetails = new UserDetails(username, username)
-                    .withPassword(apiKey);
-            try {
-                OpsgenieUser opsgenieUser = new OpsgenieUser(username, apiKey);
-                userManager.createUser(userDetails);
-                ConnectionSetupDto connectionSetupDto = new ConnectionSetupDto()
-                        .setUsername(opsgenieUser.getUsername())
-                        .setPassword(opsgenieUser.getPassword())
-                        .setServerId(getServerId().orElseThrow(() -> new ValidationException("serverId is empty!")))
-                        .setServerUrl(opsgeniePluginSettings.getServerUrl());
-                SendResult result = opsgenieClient.post(baseUrl + SETUP_ENDPOINT, apiKey, gson.toJson(connectionSetupDto));
-                if (!result.isSuccess()) {
-                    throw new Exception(result.getFailReason() +
-                            ". Created User needs to be deleted and plugin needs to be reconfigured.");
-                }
-            } catch (Exception e) {
-                deleteSettings();
-                throw new OpsgenieUserCreationFailedException(e.getMessage());
+
+        if (findUser().isPresent()) {
+            throw new OpsgenieUserCreationFailedException("User: " + username + " already exists. Please remove it first!");
+        };
+
+        if (StringUtils.isNotBlank(apiKey) && StringUtils.isNotBlank(baseUrl)) {
+            OpsgenieUser opsgenieUser = new OpsgenieUser(username, apiKey);
+            ConnectionSetupDto connectionSetupDto = new ConnectionSetupDto()
+                    .setUsername(opsgenieUser.getUsername())
+                    .setPassword(opsgenieUser.getPassword())
+                    .setServerId(getServerId().orElseThrow(() -> new ValidationException("serverId is empty!")))
+                    .setServerUrl(opsgeniePluginSettings.getServerUrl());
+            SendResult result = opsgenieClient.post(baseUrl + SETUP_ENDPOINT, apiKey, gson.toJson(connectionSetupDto));
+
+            if (result.isSuccess()) {
+                createUser(opsgeniePluginSettings);
+            } else {
+                throw new OpsgenieUserCreationFailedException(result.getFailReason());
             }
         }
     }
 
+    private Optional<ApplicationUser> findUser() {
+        ApplicationUser applicationUser = userManager.getUserByName(OpsgeniePluginSettingsManager.OPSGENIE_USERNAME);
+        if (applicationUser == null) {
+            return Optional.empty();
+        } else {
+            return Optional.of(applicationUser);
+        }
+    }
+
+    private void createUser(OpsgeniePluginSettings settings) throws OpsgenieUserCreationFailedException {
```

Comment:
Consider adding a param to update the user on Jira adapter in case of deleted user.